src/scripts/Build-Bicep.ps1

```diff
@@ -0,0 +1,189 @@
+                    # If current scope, uncomment; otherwise, skip line
+                    if ($isCurrentScopeBlock) {
+                        append ([regex]'//\s*').Replace($lines[++$i], '', 1)
+                    } else {
+                        $i++
+                    }
+                }
+            }
+            # Append standard code
+            else {
+                append $line
+            }
+ 
+            # Move to next line
+            $i++
+        }
+  
+        # Write main file
+        ./New-Directory (Join-Path $outdir $moduleName (Split-Path $Path))
+        if ($Debug) { 
+            $sb.ToString()
+        } else {
+            $sb.ToString() | Out-File (Join-Path $outdir $moduleName $Path)
+        }
+
+        # Write supporting files
+        if ($CopySupportingFiles -and -not $Debug) {
+            @('main.json', 'metadata.json', 'README.md', 'version.json') `
+            | ForEach-Object { Copy-Item (Join-Path $Module $_) (Join-Path $outdir $moduleName) }
+        }
+    }
+}
+
+# Generate module and test code
+Build-Modules main.bicep -CopySupportingFiles
+Build-Modules (Join-Path test main.test.bicep)
+
+# Copy tests to README.md examples
+if (-not $Debug) {    
+    Get-ChildItem (Join-Path $outdir * test main.test.bicep) `
+    | Where-Object { $_.Directory.Parent.Name -match "-$($dir.Name)$" } `
+    | ForEach-Object {
+        $testFile = $_
+        $sb = [System.Text.StringBuilder]::new()
+        $writingModule = $false
+        
+        Get-Content $testFile `
+        | ForEach-Object {
+            $line = $_
+            # If test comment, write example header
+            if ($line -match '^\s*//\s*Test\s') {
+                # Parse test number and description
+                $regex = [regex]::Matches($line.Trim(), '^//\s*Test\s*([0-9]+)\s*-\s*(.*)');
+                $number = $regex.Groups[1].Value
+                $text = $regex.Groups[2].Value
+    
+                [void]$sb.AppendLine().AppendLine("### Example $number").AppendLine()
+                [void]$sb.AppendLine($text).AppendLine()
+                [void]$sb.AppendLine('```bicep')
+            }
+            # If module, adjust the target destination
+            elseif ($line -match '^module\s*' -and $line -match '../main.bicep') {
+                $line = $line -replace "../main.bicep", "br/public:cost/$($testFile.Directory.Parent.Name):1.0"
+                [void]$sb.AppendLine($line)
+                $writingModule = $true
+            }
+            # If module body, append code
+            elseif ($writingModule -and -not ($line -match '^}$')) {
+                [void]$sb.AppendLine($line)
+            }
+            # If end of module, close code block
+            elseif ($writingModule -and $line -eq '}') {
+                [void]$sb.AppendLine($line).AppendLine('```')
+                $writingModule = $false
+            }
+        }
+    
+        # Append examples to README file
```

Comment:
Will this make the readme grow and grow if run frequently?

Reply:
No. We only append the examples to the generated file, so the original file is always example-less.